### PR TITLE
Enabling async functions by default

### PR DESCRIPTION
**Pull request by aneeshdk**
Enabling async functions by default. Also added one new test case to
verify the arguments.callee behavior.

### CODE REVIEW

File: test/es7/asyncawait-functionality.js [hunk 1]
```diff
@@ -1,939 +1,956 @@
...
+                } else {
+                    echo(`Test #${index} - Failure a appears to have an unexpected value a = '${result}'`);
+                }
+                if (result[1]() === 1) {
+                    echo(`Test #${index} - Success function defined in the param scope captures the param scope variable with eval in the body`);
+                } else {
+                    echo(`Test #${index} - Failure a appears to have an unexpected value in the param scope function a = '${result}'`);
+                }
+            }, err => {
+                echo(`T
```
Comment (aneeshdk) [inline test/es7/asyncawait-functionality.js]: This is the new test case added

**Comment by agarwal-sandeep** [comment]
Can we also move async debugger tests from full to core, that will make sure async functionality is working as expected in ChakraCore/JsRT debugging as well.

**Comment by aneeshdk** [comment]
Moved the debugger test cases

File: test/DebuggerCommon/async.js.dbg.baseline [hunk 2]
```diff
@@ -0,0 +1,109 @@
+[
+  {
+    "callStack": [
+      {
+        "line": 10,
+        "column": 4,
+        "sourceText": "return await af1(10)",
+        "function": "af2"
+      },
+      {
```
Comment (agarwal-sandeep) [inline test/DebuggerCommon/async.js.dbg.baseline]: In PDM based debugger baselines I see a frame here "Generator.prototype.next" why is it not here?

Comment (agarwal-sandeep) [inline test/DebuggerCommon/async.js.dbg.baseline]: Ok, it seems PDM enables library stack frames and our default value is false so ChakraCore doesn't give it. For now its fine but we may need to expose a functionality to enable this in ChakraCore in future.

Comment (akroshg) [inline test/DebuggerCommon/async.js.dbg.baseline]: Yes, nice to have that... 

---
In reply to: [82048419](https://github.com/Microsoft/ChakraCore/pull/1691#discussion_r82048419) [](ancestors = 82048419)

File: test/DebuggerCommon/async_step_over.js.dbg.baseline [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        "function": "Global code"
+      }
+    ]
+  },
+  {
+    "callStack": [
+      {
+        "line": 28,
+        "column": 0,
+        "sourceText": "p.then(result => {\r\n        if (result === 100) {\r\n            print(\"PASS\");\r\n        }\r\n    },\r\n    error => {\r\n        print(\"Failed : \" + error);\r\n    } \r\n)",
```
Comment (akroshg) [inline test/DebuggerCommon/async_step_over.js.dbg.baseline]: curious why it is not truncated. Did we change the default, or ch has different default?

File: test/es6/rlexe.xml [hunk 4]
```diff
@@ -756,19 +755,19 @@
   <test>
     <default>
       <files>default.js</files>
-      <compile-flags>-force:deferparse -ES6DefaultArgs -ES7AsyncAwait -ES6Generators -args summary -endargs</compile-flags>
+      <compile-flags>-force:deferparse -ES6DefaultArgs -ES6Generators -args summary -endargs</compile-flags>
```
Comment (akroshg) [inline test/es6/rlexe.xml]: nit: while you are here , why don't you remove other flags for this test?

**Comment by akroshg** [comment]
:shipit:

File: test/DebuggerCommon/async_step_over.js.dbg.baseline (hunk 3 above)
Comment (agarwal-sandeep) [inline test/DebuggerCommon/async_step_over.js.dbg.baseline]: sourceText is not truncated, same as PDM (callstack text) baseline behavior.

**Comment by aneeshdk** [comment]
@dotnet-bot test ubuntu_linux_test please

**Comment by aneeshdk** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_test  please
@dotnet-bot test Ubuntu ubuntu_linux_debug_static please

**Pull request merged**